process/process_test.go

```diff
@@ -72,3 +75,41 @@ func TestTerminateAllFn(t *testing.T) {
 	}
 	terminateAll(fn, "", time.Millisecond, testLog)
 }
+
+// process returns this test process' path that is running
+func process(t *testing.T) (int, string) {
+	pid := os.Getpid()
+	proc, err := findProcessWithPID(pid)
+	require.NoError(t, err)
+	path, err := proc.Path()
+	require.NoError(t, err)
+	return pid, path
+}
+
+func TestFindProcessTest(t *testing.T) {
+	if runtime.GOOS == "linux" {
+		t.Skip("Unsupported until we have process path on linux")
+	}
+	pid, path := process(t)
+	procs, err := FindProcesses(path, 0, 0, testLog)
+	require.NoError(t, err)
+	require.True(t, len(procs) == 1)
+	assert.Equal(t, pid, procs[0].Pid())
+}
+
+func TestFindProcessWait(t *testing.T) {
+	path := "/bin/echo"
+	go func() {
+		time.Sleep(10 * time.Millisecond)
+		cmd := exec.Command(path, "hi")
+		err := cmd.Start()
+		require.NoError(t, err)
+	}()
```

Comment:
maybe if it accepts a channel it can be tested with github.com/jonboulle/clockwork